src/esp/sensor/VisualSensor.cpp

```diff
@@ -22,5 +24,34 @@ void VisualSensor::bindRenderTarget(gfx::RenderTarget::uptr&& tgt) {
   tgt_ = std::move(tgt);
 }
 
+void VisualSensor::visualizeObservation(gfx::SensorInfoVisualizer& visualizer,
+                                        float depthScaling) {
+  auto sensorType = this->spec_->sensorType;
+  CORRADE_ASSERT(
+      sensorType == SensorType::Depth,
+      "CameraSensor::visualizeObservation: sensor type is not supported.", );
+
+  // prepare: setup and clear framebuffer
+  visualizer.prepareToDraw(framebufferSize());
+  switch (sensorType) {
+    case SensorType::Depth: {
+      // setup the shader
+      Mn::Resource<Mn::GL::AbstractShaderProgram, gfx::DepthVisualizerShader>
+          shader = visualizer.getShader<gfx::DepthVisualizerShader>(
+              gfx::SensorInfoType::Depth);
+      shader->bindDepthTexture(renderTarget().getDepthTexture())
+          .setDepthUnprojection(*depthUnprojection())
+          .setDepthScaling(depthScaling);
+      // draw to the framebuffer
+      visualizer.draw(shader.key());
+    } break;
+
+    default:
+      // Have to had this default, otherwise the clang-tidy will be pissed off,
+      // and will not let me pass
+      break;
```

Comment:
:laughing: 

what was the error it gave you? this seems silly

Reply:
The CI test failed directly in the clang-tidy section.
It says missing code for the other enum terms. 
And look at this [long list](https://github.com/facebookresearch/habitat-sim/blob/master/src/esp/sensor/Sensor.h#L20-L32):   
  None = 0,
  Color = 1,
  Depth = 2,
  Normal = 3,
  Semantic = 4,
  Path = 5,
...

Reply:
@Skylion007 : the "default" a bad concept.

Reply:
Oh, right, I didn't realize there's a lot more values. In that case either enumerating them all or adding a `default`. In this case the default probably makes sense.